sending/src/main/java/org/kin/sending/view/ContactDialogFragment.java

```diff
@@ -0,0 +1,220 @@
+package org.kin.sending.view;
+
+import android.content.ClipboardManager;
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.constraint.Group;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.Gravity;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.kin.sending.R;
+import org.kin.sending.presenter.ContactDialogPresenter;
+import org.kin.sending.presenter.ContactDialogPresenterImpl;
+import org.kin.sendkin.core.view.BaseWideDialogFragment;
+import org.kin.sendkin.core.view.SendKinEditText;
+import org.kin.sendkin.core.view.Utils;
+
+import java.util.UUID;
+
+public class ContactDialogFragment extends BaseWideDialogFragment implements ContactDialogView {
+    public static final String TAG = ContactDialogFragment.class.getSimpleName();
+
+    private static final String KEY_ID = "KEY_ID";
+    private ImageView deleteIcon;
+    private TextView title, addressNotValidWarning, saveBtn, deleteSubtitle;
+    private ContactDialogPresenter presenter;
+    private SendKinEditText nameInput, addressInput;
+    private Group nameAddressLayout;
+
+    private TextWatcher nameWatcher = new TextWatcher() {
```

Comment:
Don't think you need a watcher because validation is only after pressing save.

Reply:
good point